src/boot/efi/shim.c

```diff
@@ -0,0 +1,255 @@
+
+                result = shim_lock->shim_verify(data, size);
+                if (result == EFI_SUCCESS)
+                        return TRUE;
+
+                Print(L"Security Validation Failed: %r \n", result);
+                uefi_call_wrapper(BS->Stall, 1, 3 * 1000 * 1000);
+        }
+
+        return FALSE;
+}
+
+BOOLEAN secure_boot_enabled(void) {
+        CHAR8 *b;
+        UINTN size;
+        BOOLEAN result;
+
+        if (efivar_get_raw(&GLOBAL_GUID, L"SecureBoot", &b, &size) == EFI_SUCCESS) {
+                result = *b > 0;
+                FreePool(b);
+                return result;
+        } else {
+                return FALSE;
+        }
+}
+
+/*
+ * See the UEFI Platform Initialization manual (Vol2: DXE) for this
+ */
+struct _EFI_SECURITY2_PROTOCOL;
+struct _EFI_SECURITY_PROTOCOL;
+struct _EFI_DEVICE_PATH_PROTOCOL;
+typedef struct _EFI_SECURITY2_PROTOCOL EFI_SECURITY2_PROTOCOL;
+typedef struct _EFI_SECURITY_PROTOCOL EFI_SECURITY_PROTOCOL;
+typedef struct _EFI_DEVICE_PATH_PROTOCOL EFI_DEVICE_PATH_PROTOCOL;
+
+typedef EFI_STATUS (EFIAPI *EFI_SECURITY_FILE_AUTHENTICATION_STATE) (
+        const EFI_SECURITY_PROTOCOL *This,
+        UINT32 AuthenticationStatus,
+        const EFI_DEVICE_PATH_PROTOCOL *File
+);
+
+typedef EFI_STATUS (EFIAPI *EFI_SECURITY2_FILE_AUTHENTICATION) (
+        const EFI_SECURITY2_PROTOCOL *This,
+        const EFI_DEVICE_PATH_PROTOCOL *DevicePath,
+        VOID *FileBuffer,
+        UINTN FileSize,
+        BOOLEAN  BootPolicy
+);
+
+struct _EFI_SECURITY2_PROTOCOL {
+        EFI_SECURITY2_FILE_AUTHENTICATION FileAuthentication;
+};
+
+struct _EFI_SECURITY_PROTOCOL {
+        EFI_SECURITY_FILE_AUTHENTICATION_STATE  FileAuthenticationState;
+};
+
+static EFI_SECURITY_FILE_AUTHENTICATION_STATE esfas = NULL;
+static EFI_SECURITY2_FILE_AUTHENTICATION es2fa = NULL;
+
+/*
+ * Perform shim/MOK and Secure Boot authentication on a binary that's already been
+ * loaded into memory. This function does the platform SB authentication first
+ * but preserves its return value in case of its failure, so that it can be
+ * returned in case of a shim/MOK authentication failure. This is done because
+ * the SB failure code seems to vary from one implementation to another, and I
+ * don't want to interfere with that at this time.
+ */
+static EFIAPI EFI_STATUS security2_policy_authentication (
+        const EFI_SECURITY2_PROTOCOL *This,
+        const EFI_DEVICE_PATH_PROTOCOL *DevicePath,
+        VOID *FileBuffer,
+        UINTN FileSize,
+        BOOLEAN  BootPolicy
+) {
+        EFI_STATUS Status;
+
+        /* Chain original security policy */
+        Status = uefi_call_wrapper(es2fa, 5, This, DevicePath, FileBuffer, FileSize, BootPolicy);
+
+        /* if OK, don't bother with MOK check */
+        if (Status == EFI_SUCCESS)
+                return Status;
+
+        if (shim_validate(FileBuffer, FileSize)) {
+                return EFI_SUCCESS;
+        } else {
+                return Status;
+        }
```

Comment:
no {} around single-line if blocks....

Also, given that you return in the "if" branch, you can drop the "else" branch, and just place the "return Status" in the normal control flow